### ISSUE TITLE
User defined pipes vs builtin pipes

### ISSUE THREAD

**Issue by kemsky** [angular/angular#35502]
# 🐞 bug report

### Affected Package

 The issue is caused by package 
@angular/common

### Is this a regression?

I think it is.

### Description

User can define his own pipe with name `currency` for example. Previously angular used *user* pipe instead of builtin. Ivy does opposite silently.

## 🔬 Minimal Reproduction

You can copy & paste any pipe from `number_pipe.ts` modify its transform method and see that pipe will be ignored.

## 🔥 Exception or Error

No error, but it should be.

## 🌍  Your Environment

**Angular Version:**
<pre><code>

Angular: 9.0.1
... common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.803.9 (cli-only)
@angular-devkit/core         8.3.9 (cli-only)
@angular-devkit/schematics   8.3.9 (cli-only)
@angular/cdk                 9.0.0
@schematics/angular          8.3.9 (cli-only)
@schematics/update           0.803.9 (cli-only)
rxjs                         6.5.4
typescript                   3.7.5
webpack                      4.41.6
</code></pre>

**Comment by sacgrover** [comment]
I doubt if it is possible in angular 9 as Injectable() have been removed from build-in pipes.

**Comment by AndrewKushnir** [comment]
Hi @kemsky, thanks for reporting the issue. We performed the investigation and it looks like the problem only happens in case custom Pipe is defined as a part of `declarations` array of an NgModule, i.e.:

```
@NgModule({
  imports: [CommonModule],
  declarations: [CustomNumberPipe],
})
class SomeModule {}
```
However if you declare a pipe in a separate module and import it, the framework works as expected, i.e.:

```
@NgModule({
  declarations: [CustomNumberPipe],
  exports: [CustomNumberPipe],
})
class CustomNumberPipeModule {}

@NgModule({
  imports: [CommonModule, CustomNumberPipeModule],
})
class SomeModule {}
```

I've created a draft PR https://github.com/angular/angular/pull/35850 to make sure the `declarations` field has higher priority in the first case that I described.

Could you please confirm that the Pipes/NgModule layout that you have in your app is similar to the first scenario that I described?

Thank you.

**Comment by AndrewKushnir** [comment]
@kemsky if you get a chance, it'd be great if you could also help verify that the changes from PR #35850 resolve the issue. Note: PR still requires code review and testing, so the final implementation might be changed.

You can follow an instruction in [this document on how to use PR artifacts](https://github.com/angular/angular/blob/master/docs/DEVELOPER.md#archives-for-each-package) in your app. An archive with updated `@angular/core` package that contains changes from the PR is available [here](https://642504-24195339-gh.circle-artifacts.com/0/angular/core-pr35850-ccd466f934.tgz) and updated `@angular/compiler-cli` can be found [here](https://642504-24195339-gh.circle-artifacts.com/0/angular/compiler-cli-pr35850-ccd466f934.tgz) (please make sure you update other `@angular` packages to `9.0.4` to avoid version conflicts). **Note**: the mentioned archives with updated `@angular/core` and `@angular/compiler-cli` packages are for **testing purposes only** and not suitable for production usage (that archive will be removed once PR is merged to master or closed).

Thank you.

**Comment by kemsky** [comment]
@AndrewKushnir, in our case application has single module so it probably explains everything. I'll try test your PR, thanks.
